docs/concepts/glossary.md

```diff
@@ -78,6 +81,9 @@ A view is the result of a SQL query on a database.
 ## Virtual Environments
 SQLMesh's unique approach to environment that allows it to provide both environment isolation and the ability to share tables across environments. This is done in a way to ensure data consistency and accuracy. See [plan application](plans.md#plan-application) for more information.
 
+## Virtual Layer
+The virtual layer is a conceptual framework managed by SQLMesh that provides an abstraction layer over the physical layer and physical data storage. Unlike the physical layer, where data is actually stored in tables and views, the virtual layer exists purely as a set of views, derived from the model definitions and environments, that expose the underlying physical layer. While typically transparent to end users, the virtual layer is the layer that users interact with when querying data. 
```

Comment:
The virtual layer is SQLMesh's abstraction layer over the [physical layer and physical data storage](#physical-layer). While the physical layer consists of tables where data is actually stored, the virtual layer consists of views that expose tables in the underlying physical layer. When a `plan` updates the models in an environment, SQLMesh automatically updates the environment's virtual layer views to point to the correct underlying physical table. Most users should only interact with the virtual layer when building models or querying data.

Reply:
Interesting. I guess in my definition, it was the more abstract layer that encompasses all the environments' virtual part whereas in yours, the virtual layer is part of an environment.

Reply:
That's also how I think of it, but I thought it might be useful to clarify that each model name can have one view per environment, where the views point to different underlying physical tables. I'm now doubting that and think the environment-specific aspect is too detailed for a glossary definition. 

I updated my comment above to remove that sentence and I think it reads better - wdyt?

Reply:
I like it!